internal/blueprint/filesystem_customizations.go

```diff
@@ -22,19 +23,57 @@ func (fsc *FilesystemCustomization) UnmarshalTOML(data interface{}) error {
 		return fmt.Errorf("TOML unmarshal: mountpoint must be string, got %v of type %T", d["mountpoint"], d["mountpoint"])
 	}
 
+	var size uint64
+	var minsize uint64
+
 	switch d["size"].(type) {
 	case int64:
-		fsc.MinSize = uint64(d["size"].(int64))
+		size = uint64(d["size"].(int64))
 	case string:
-		size, err := common.DataSizeToUint64(d["size"].(string))
+		s, err := common.DataSizeToUint64(d["size"].(string))
 		if err != nil {
 			return fmt.Errorf("TOML unmarshal: size is not valid filesystem size (%w)", err)
 		}
-		fsc.MinSize = size
+		size = s
+	case nil:
+		size = 0
 	default:
 		return fmt.Errorf("TOML unmarshal: size must be integer or string, got %v of type %T", d["size"], d["size"])
 	}
 
+	switch d["minsize"].(type) {
+	case int64:
+		minsize = uint64(d["minsize"].(int64))
+	case string:
+		s, err := common.DataSizeToUint64(d["minsize"].(string))
+		if err != nil {
+			return fmt.Errorf("TOML unmarshal: minsize is not valid filesystem size (%w)", err)
+		}
+		minsize = s
+	case nil:
+		minsize = 0
+	default:
+		return fmt.Errorf("TOML unmarshal: minsize must be integer or string, got %v of type %T", d["minsize"], d["minsize"])
+	}
+
+	if size == 0 && minsize == 0 {
+		return fmt.Errorf("TOML unmarshal: minsize must be specified, got %v", minsize)
+	}
+
+	if size > 0 && minsize == 0 {
+		fsc.MinSize = size
+		return nil
+	}
+
+	if size == 0 && minsize > 0 {
+		fsc.MinSize = minsize
+		return nil
+	}
+
+	if size > 0 && minsize > 0 {
+		return fmt.Errorf("TOML unmarshal: size and minsize cannot both be set, got %v & %v", size, minsize)
```

Comment:
We could mention that one is an alias for the other or that the two options have the same meaning.  The current message might imply that they're in conflict but that they have different meaning (for example, one could think that `size` sets a strict exact size for the partition whereas `minsize` sets a flexible size with a lower bound).

Something like
```
size and minsize cannot be both set (size is an alias for minsize)
```
but feel free to rephrase.

Reply:
Good insight, wording seems good to me too :)